pkg/scheduler/objects/application.go

```diff
@@ -177,13 +177,28 @@ func (sa *Application) HandleApplicationEvent(event applicationEvent) error {
 	return err
 }
 
-func (sa *Application) OnStateChange(event *fsm.Event) {
+func (sa *Application) HandleApplicationEventWithInfo(event applicationEvent, eventInfo string) error {
+	err := sa.stateMachine.Event(event.String(), sa, eventInfo)
+	// handle the same state transition not nil error (limit of fsm).
+	if err != nil && err.Error() == noTransition {
+		return nil
+	}
+	return err
+}
+
+func (sa *Application) OnStateChange(event *fsm.Event, eventInfo string) {
 	updatedApps := make([]*si.UpdatedApplication, 0)
+	var message string
+	if len(eventInfo) == 0 {
+		message = fmt.Sprintf("Status change triggered by the event: %s", event.Event)
+	} else {
+		message = eventInfo
+	}
```

Comment:
feel free to remove the message "Status change triggered by the event", this is not informative, I think we can ignore it.

Reply:
Cool I feel the same. Removed it